src/window_varlist.h

```diff
@@ -43,52 +43,136 @@ class Window_VarList : public Window_Command
 	 *
 	 * @param commands commands to display.
 	 */
-	Window_VarList(std::vector<std::string> commands);
+	Window_VarList();
 	~Window_VarList() override;
 
+	void Update() override;
+
 	/**
 	 * UpdateList.
 	 * 
 	 * @param first_value starting value.
 	 */
 	void UpdateList(int first_value);
 
+	void UpdateCursorRect() override;
+
 	/**
 	 * Refreshes the window contents.
 	 */
 	void  Refresh();
 
+	int GetItemIndex() const;
+
+	void SetItemIndex(int index);
+
 	/**
 	 * Indicate what to display.
 	 *
 	 * @param mode the mode to set.
-	 * @param max_length_strings maximum size for items (Used for Strings)
 	 */
-	void SetMode(Mode mode, int max_length_strings = 0);
+	void SetMode(Mode mode);
 
 	/**
 	 * Returns the current mode.
 	 */
 	Mode GetMode() const;
 
+	void SetShowDetail(bool show_detail);
+
+	bool GetShowDetail() const;
+
+	void SetItemText(unsigned index, std::string_view text);
+
+	static constexpr std::string_view GetPrefix(Mode mode);
+	static constexpr int GetItemCount(Mode mode, bool show_detail);
+	static constexpr int GetDigitCount(Mode mode);
+
+	static int GetNumElements(Mode mode);
+
+	int GetItemCount() const;
+	int GetDigitCount() const;
 private:
 
+	void DrawItem(int index, Font::SystemColor color);
+
 	/**
 	 * Draws the value of a variable standing on a row.
 	 *
 	 * @param index row with the var
 	 */
 	void DrawItemValue(int index);
 
+	void DrawStringVarItem(int index, int y);
+
+	std::vector<std::string> items;
 	Mode mode = eNone;
-	int first_var = 0, max_length_strings = 0;
+	int first_var = 0;
+	bool show_detail = false;
+	bool suspend_cursor_refresh = false;
 
 	bool DataIsValid(int range_index);
-
 };
 
+constexpr std::string_view Window_VarList::GetPrefix(Mode mode) {
+	switch (mode) {
+		case eSwitch:
+			return "Sw";
+		case eVariable:
+			return "Vr";
+		case eItem:
+			return "It";
+		case eTroop:
+			return "Bt";
+		case eMap:
+			return "Mp";
+		case eCommonEvent:
+			return "Ce";
+		case eMapEvent:
+			return "Me";
+		case eString:
+			return "St";
+		default:
+			break;
```

Comment:
move the ``assert(false)`` into the ``default:`` followed by ``return {};``. Yeah, compiler is picky here because assert is removed in release builds.